pkg/generate/app/componentref.go

```diff
@@ -191,6 +222,7 @@ func (r PerfectMatchWeightedResolver) Resolve(value string) (*ComponentMatch, er
 	case 0:
 		return nil, ErrNoMatch{value: value}
 	case 1:
+		// This is an inexact match so our PerfectMatchWeightedResolver is not returning only perfect matches...
```

Comment:
You can be inexact but still perfect.  DockerHub images are inexact, but perfect.